src/index.js

```diff
@@ -0,0 +1,187 @@
+import { didKeyType, publicKeyBytesToDid, verify } from './did.js'
+import { base64url, utf8 } from './encoding.js'
+import {
+  serialize,
+  deserialize,
+  jwtAlgorithm,
+  isExpired,
+  isTooEarly,
+} from './utils.js'
+
+const TYPE = 'JWT'
+const VERSION = '0.8.0'
+
+export { KeyPair } from './keypair.js'
+
+/**
+ * @param {import("./types").BuildParams} params
+ */
+export async function build(params) {
+  const keypair = params.issuer
+  const didStr = publicKeyBytesToDid(keypair.publicKey)
+  const payload = buildPayload({
+    ...params,
+    issuer: didStr,
+  })
+  return sign(payload, keypair)
+}
+
+/**
+ * Build parts
+ *
+ * @param {import('./types').BuildPayload} params
+ */
+function buildPayload(params) {
+  const {
+    issuer,
+    audience,
+    capabilities = [],
+    lifetimeInSeconds = 30,
+    expiration,
+    notBefore,
+    facts,
+    proofs = [],
+    // addNonce = false,
+  } = params
+
+  // Timestamps
+  const currentTimeInSeconds = Math.floor(Date.now() / 1000)
+  const exp = expiration || currentTimeInSeconds + lifetimeInSeconds
+
+  /** @type {import('./types').UcanPayload} */
+  const payload = {
+    aud: audience,
+    att: capabilities,
+    exp,
+    fct: facts,
+    iss: issuer,
+    nbf: notBefore,
+    // nnc: addNonce ? util.generateNonce() : undefined,
+    prf: proofs,
+  }
+
+  return payload
+}
+
+/**
+ * Generate UCAN signature.
+ *
+ * @param {import("./types").UcanPayload<string>} payload
+ * @param {import("./keypair.js").KeyPair} keypair
+ */
+export async function sign(payload, keypair) {
```

Comment:
I wonder the sign function should get the header as a parameter rather than hardcoding it

Reply:
its best to hardcode for now, we need to control the header because ucan are not generic jwts.

Reply:
Good point, but I wonder if that should be controlled by the [build](https://github.com/nftstorage/ucan.storage/pull/3/files/3677eca2683621cea8e9d081ce3e99f0b408a39d#diff-bfe9874d239014961b1ae4e89875a6155667db834a410aaaa2ebe3cf89820556R26).

Basically `sign` should accept a header, payload, and the key pair.

Honestly not a big deal, just capturing thoughts while I go through the code.

Reply:
we should make an issue on this